File: backend/src/main/java/com/employee/backend/service/employee/EmployeeService.java
package com.employee.backend.service.employee;

import com.employee.backend.dto.request.EmployeeRequest;
import com.employee.backend.dto.response.EmployeeResponse;

import java.util.List;

public interface EmployeeService {
    List<EmployeeResponse> findAll();
    EmployeeResponse findById(Long id);
    EmployeeResponse create(EmployeeRequest request);
    EmployeeResponse update(Long id, EmployeeRequest request);
    void delete(Long id);
}

File: frontend/src/app/core/services/employee.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse, Employee, EmployeeRequest } from '../models';
 
@Injectable({ providedIn: 'root' })
export class EmployeeService {
  private readonly API = 'http://localhost:8080/api/employees';
 
  constructor(private http: HttpClient) {}
 
  getAll(): Observable<ApiResponse<Employee[]>> {
    return this.http.get<ApiResponse<Employee[]>>(this.API);
  }
 
  getById(id: number): Observable<ApiResponse<Employee>> {
    return this.http.get<ApiResponse<Employee>>(`${this.API}/${id}`);
  }
 
  create(request: EmployeeRequest): Observable<ApiResponse<Employee>> {
    return this.http.post<ApiResponse<Employee>>(this.API, request);
  }
 
  update(id: number, request: EmployeeRequest): Observable<ApiResponse<Employee>> {
    return this.http.put<ApiResponse<Employee>>(`${this.API}/${id}`, request);
  }
 
  delete(id: number): Observable<ApiResponse<void>> {
    return this.http.delete<ApiResponse<void>>(`${this.API}/${id}`);
  }
}
 